src/sentry/models/externalactor.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from django.db import models
+
+from sentry.db.models import BoundedPositiveIntegerField, DefaultFieldsModel, FlexibleForeignKey
+from sentry.types.integrations import ExternalProviders
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalActor(DefaultFieldsModel):
+    __core__ = False
+
+    actor = FlexibleForeignKey("sentry.Actor", db_index=True, on_delete=models.CASCADE)
+    organization = FlexibleForeignKey("sentry.Organization")
+    integration = FlexibleForeignKey("sentry.Integration", null=True)
+    provider = BoundedPositiveIntegerField(
+        choices=(
+            (ExternalProviders.EMAIL, "email"),
+            (ExternalProviders.SLACK, "slack"),
+            (ExternalProviders.MSTEAMS, "msteams"),
+            (ExternalProviders.PAGERDUTY, "pagerduty"),
+            (ExternalProviders.GITHUB, "github"),
+            (ExternalProviders.GITLAB, "gitlab"),
+        ),
+    )
+
+    # external name => display name i.e. username, team name, channel name
+    # external id => unique identifier i.e user id, channel id
+    external_name = models.TextField()
```

Comment:
maybe dumb question but why do we need both `external_name` and `external_id`? For notifications in Slack I believe we'll just need to store the IM channel ID, right? Is it for messaging teams in a channel so we'd need the channel ID and team name for the @?

Reply:
@ceorourke we need the GH/GL username or teamnames to be able to map the codeowners files. We won't have and don't need the `external_id` for that. It seems to me that if we have a slack ID we could easily get the name so I assumed it won't be a problem to have the external_name be non nullable but maybe that's not a fair assumption

Reply:
@MeredithAnya hmm yeah I'm trying to think through how we'll use this table (and make sure it works for both of our projects). For example I thought we could migrate data from the `Identity` table since we have some Sentry user IDs and Slack IDs already but we don't store the name there ... but maybe that can wait until we update permissions and we can do an API call to grab and store the name. That being said I'm not sure we need the name? Since like I said you DM a user by passing their user ID (no anchor tag unfortunately but look at "Post to a direct message channel
' in https://api.slack.com/methods/chat.postMessage#channels). What do you think @mgaeta ?